lib/react/rails/engine.rb

```diff
@@ -2,7 +2,11 @@ module React
   module Rails
     class Engine < ::Rails::Engine
       initializer "react_rails.setup_engine", :group => :all do |app|
-        app.assets.register_engine '.jsx', React::JSX::Template
+        if app.assets.nil?
+          Sprockets.register_engine  '.jsx', React::JSX::Template
+        else
+          app.assets.register_engine '.jsx', React::JSX::Template
+        end
```

Comment:
Actually its harder to read that way and make the intent clear why we are doing it.